gateway-server/src/test/java/org/apache/knox/gateway/session/control/ConcurrentSessionVerifierTest.java

```diff
@@ -106,25 +110,25 @@ public void userIsNotPrivileged() {
   }
 
   @Test
-  public void privilegedLimitIsZero() {
+  public void privilegedLimitIsZero() throws ServiceLifecycleException {
     GatewayConfig config = mockConfig(new HashSet<>(Arrays.asList("admin")), new HashSet<>(Arrays.asList("tom", "guest")), 0, 2);
-    verifier.init(config);
+    verifier.init(config, options);
 
     Assert.assertFalse(verifier.verifySessionForUser("admin"));
   }
 
   @Test
-  public void nonPrivilegedLimitIsZero() {
+  public void nonPrivilegedLimitIsZero() throws ServiceLifecycleException {
     GatewayConfig config = mockConfig(new HashSet<>(Arrays.asList("admin")), new HashSet<>(Arrays.asList("tom", "guest")), 3, 0);
-    verifier.init(config);
+    verifier.init(config, options);
 
     Assert.assertFalse(verifier.verifySessionForUser("tom"));
   }
 
   @Test
-  public void sessionsDoNotGoToNegative() {
+  public void sessionsDoNotGoToNegative() throws ServiceLifecycleException {
     GatewayConfig config = mockConfig(new HashSet<>(Arrays.asList("admin")), new HashSet<>(Arrays.asList("tom", "guest")), 2, 2);
-    verifier.init(config);
+    verifier.init(config, options);
 
     Assert.assertNull(verifier.getUserConcurrentSessionCount("admin"));
```

Comment:
Why does this return null instead of zero?

Reply:
Originally it returned 0, but me and @smolnar82 reviewed the code and decided to return null instead. It was in the previous pull request: https://github.com/apache/knox/pull/608/commits/af16b58d8c3bcbd9c40841b7c92416ab0bff97f6#diff-cf1fb21462bed96d5f2087f2729a659a2b4141b20536e114a72b24f9d82d4356.
Unfortunately, we talked via Zoom so i can not remember the exact reason, but right now i can't see any reason why returning 0 would not be good.

Reply:
As far as I remember, It was related to removing the session info from the map altogether so that it won't take any space in the Map when the counter goes zero, isn't it?

Reply:
Yes, because originally there were no tokens stored and we didn't have to count the not expired ones, so we just returned the value from the HashMap for the user, which returned null if the user was not in the HashMap. Now we have to count the not expired tokens so this function might be unnecessary, we could add countValidTokensForUser(username) function package visibility and use it in the test instead of getUserConcurrentSession(username).

Reply:
Whether or not the count is stored in the map (implementation detail), the contract of the method could be to return zero. If there is no entry in the map, that also means there are zero sessions, correct?

Reply:
The users who aren't configured for either of the groups could have sessions and they won't have entry in the map, but i think too that the method could return zero.